src/Elsa.CustomActivities/Activities/Common/WeightedCheckboxModel.cs

```diff
@@ -8,8 +8,17 @@ namespace Elsa.CustomActivities.Activities.Common
 {
     public class WeightedCheckboxModel
     {
+        public IDictionary<string, WeightedCheckboxGroup> Groups { get; set; } = new Dictionary<string, WeightedCheckboxGroup>();
+
+    }
+    public class WeightedCheckboxGroup
+    {
+        public string GroupIdentifier { get; set; } = null!;
         public ICollection<WeightedCheckboxRecord> Choices { get; set; } = new List<WeightedCheckboxRecord>();
+        public decimal? MaxGroupScore { get; set; }
+        public List<decimal>? GroupArrayScore { get; set; }
+
     }
 
-    public record WeightedCheckboxRecord(string Identifier, string Answer, bool IsSingle, string Score, bool IsPrePopulated);
+    public record WeightedCheckboxRecord(string Identifier, string Answer, bool IsSingle, decimal Score, bool IsPrePopulated);
```

Comment:
Should this be a string?

Reply:
no, I don't think so, it should be decimal